### PR TITLE
Client/server code for MinIO testing

### PR DESCRIPTION
**Pull request by ustiugov**
- [x] add a function for minio testing (server)
- [x] add a Go client for invoking that function
- [x] add scripts that test the corresponding workflow using Docker containers and in Knative
- [x] include the workflow into the CRI tests

The function is a gRPC server that saves a payload to the MinIO server that is deployed in vHive, then immediately loads it from MinIO.

Signed-off-by: Dmitrii Ustiugov <dmitrii.ustiugov@epfl.ch>

### CODE REVIEW

**Comment by ustiugov** [comment]
@amariucaitheodor this test works but I'd like to add this test to the CI before merging

File: function-images/tests/save_load_minio/Makefile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+all: local push_docker
+
+.PHONY: proto_gen
+proto_gen:
+	GO111MODULE="on" go get google.golang.org/protobuf/cmd/protoc-gen-go \
```
Comment (ustiugov) [inline function-images/tests/save_load_minio/Makefile:27]: I anticipate that protoc may be useful for other functions. Please move its installation to install_go.sh

File: function-images/tests/save_load_minio/Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+build_local:
+	mkdir -p bins
+	cd client && go build -o ../bins/client
+	cd server && go build -o ../bins/server
+
+push_docker: build_docker
+	docker tag save_load_minio_test:latest vhiveease/save_load_minio_test:latest
+	docker push vhiveease/save_load_minio_test:latest
+
+build_docker: proto_gen
```
Comment (ustiugov) [inline function-images/tests/save_load_minio/Makefile:46]: please add to the docker images build test

File: function-images/tests/save_load_minio/proto_gen/go.mod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module proto_gen
+
+go 1.14
```
Comment (ustiugov) [inline function-images/tests/save_load_minio/proto_gen/go.mod:3]: 1.15 please

File: function-images/tests/save_load_minio/Makefile (hunk 1 above)
Comment (shyamjesal) [reply function-images/tests/save_load_minio/Makefile:27]: we already have protoc in the container. Do we still require `protoc-gen-go` separately?

File: function-images/tests/save_load_minio/proto_gen/go.mod (hunk 3 above)
Comment (shyamjesal) [reply function-images/tests/save_load_minio/proto_gen/go.mod:3]: done

File: function-images/tests/save_load_minio/Makefile (hunk 2 above)
Comment (shyamjesal) [reply function-images/tests/save_load_minio/Makefile:46]: done

**Review by ustiugov — COMMENTED** [review]
all good, approve and merge please

**Comment by ustiugov** [comment]
@shyamjesal please don't forget to delete the branches that are merged

**Pull request merged**